ksql-common/src/main/java/io/confluent/ksql/util/SchemaUtil.java

```diff
@@ -363,4 +375,40 @@ public static Schema ensureOptional(final Schema schema) {
         .build();
   }
 
+
+  public static boolean compareSchemaTypes(final Schema arg1, final Schema arg2) {
+    if (arg2 == null) {
+      return arg1.isOptional();
+    }
+
+    // we require a custom equals method that ignores certain values (e.g.
+    // whether or not the schema is optional, and the documentation)
+    return Objects.equals(arg1.type(), arg2.type())
+        && CUSTOM_SCHEMA_EQ.getOrDefault(arg1.type(), (a, b) -> true).test(arg1, arg2)
+        && Objects.equals(arg1.version(), arg2.version())
+        && Objects.deepEquals(arg1.defaultValue(), arg2.defaultValue());
+  }
+
+  private static boolean mapEquals(final Schema mapA, final Schema mapB) {
+    return Objects.equals(mapA.keySchema(), mapB.keySchema())
+        && Objects.equals(mapA.valueSchema(), mapB.valueSchema());
+  }
+
+  private static boolean arrayEquals(final Schema arrayA, final Schema arrayB) {
+    return Objects.equals(arrayA.valueSchema(), arrayB.valueSchema());
+  }
+
+  private static boolean structEquals(final Schema structA, final Schema structB) {
+    return structA.fields().isEmpty()
+        || structB.fields().isEmpty()
+        || Objects.equals(structA.fields(), structB.fields());
+  }
+
+  private static boolean bytesEquals(final Schema bytesA, final Schema bytesB) {
+    // from a UDF parameter perspective, all decimals are the same
```

Comment:
nit: since we moved this into SchemaUtil we should update the comment to say "from a Java schema perspective..."